### PR TITLE
EdgeR complex designs

### PR DESCRIPTION
**Pull request by pcm32**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

This PR replaces #2490 to avoid the needed rebase on master that was introducing large conflicts that was difficult to decide.

### CODE REVIEW

**Comment by bgruening** [comment]
@mblue9 can you please look at this?

File: tools/edger/edger.xml [hunk 1]
```diff
@@ -175,9 +175,9 @@ cp '$outReport.files_path'/*.tsv output_dir/
...
-            <param name="contrast" type="text" label="Contrast of Interest" help="Names of two groups to compare separated by a hyphen e.g. Mut-WT. If the order is Mut-WT the fold changes in the results will be up/down in Mut relative to WT. If you have more than one contrast enter each separately using the Insert Contrast button below. For more info, see Chapter 8 in the limma User's guide: https://www.bioconductor.org/packages/release/bioc/vignettes/limma/inst/doc/usersguide.pdf">
+            <param name="contrast" type="text" label="Contrast of Interest" help="Names of two groups to compare separated by a hyphen e.g. Mut-WT. If the order is Mut-WT the fold changes in the results will be up/down in Mut relative to WT. If you have more than one contrast enter each separately using the Insert Contrast button below. For nested contrasts, such as (MT.t1-MT.t0)-(WT.t1-WT.t0) you can use parenthesis. For more info, see Chapter 8 in the limma User's guide: https://www.bioconductor.org/packages/release/bioc/vignettes/limma/inst/doc/usersguide.pdf or https://bioconductor.org/packages/release/bioc/vignettes/edgeR/inst/doc/edgeRUsersGuide.pdf page 36 for nested comparisons.">
```
Comment (mblue9) [inline tools/edger/edger.xml]: afaik this is the syntax for difference of differences, nested would be the colon syntax e.g Condition:Time see pages 36/37 here:
https://www.bioconductor.org/packages/release/bioc/vignettes/edgeR/inst/doc/edgeRUsersGuide.pdf

Maybe use the same wording as in the limma tool for consistency? Use same wording as in the limma tool for consistency?
e.g. here https://github.com/galaxyproject/tools-iuc/blob/master/tools/limma_voom/limma_voom.xml#L241

File: tools/edger/edger.xml [hunk 2]
```diff
@@ -175,9 +175,9 @@ cp '$outReport.files_path'/*.tsv output_dir/
...
                 <validator type="empty_field" />
-                <validator type="regex" message="Please only use letters, numbers or underscores">^[\w-]+$</validator>
+                <validator type="regex" message="Please only use letters, numbers, parenthesis or underscores">^[\w\-()]+$</validator>
```
Comment (mblue9) [inline tools/edger/edger.xml]:
```suggestion
                <validator type="regex" message="Please only use letters, numbers, parentheses or underscores">^[\w\-()]+$</validator>
```

**Comment by pcm32** [comment]
Thanks for the review @mblue9, I think that the above commits should target what you mentioned (hopefully I didn't miss anything).

**Pull request merged**